src/L.Control.Locate.js

```diff
@@ -514,7 +514,10 @@ You can find the project at: https://github.com/domoritz/leaflet-locatecontrol
                 this._event = undefined;  // clear the current location so we can get back into the bounds
                 this.options.onLocationOutsideMapBounds(this);
             } else {
-                if (this.options.keepCurrentZoomLevel) {
+                if (this.options.keepCurrentZoomLevel && (
+                    typeof this.options.keepCurrentZoomLevel == "boolean" ||
+                    (typeof this.options.keepCurrentZoomLevel == "object" && this._map.getZoom() <= this.options.keepCurrentZoomLevel[1] && this._map.getZoom() >= this.options.keepCurrentZoomLevel[0])
```

Comment:
Can you extract this into a variable `zoomOutsideRange` and then write `if (this.options.keepCurrentZoomLevel || zoomOutsideRange)`?

Reply:
Not sure what mean, make something like 

      var zoomOutsideRange = typeof this.options.keepCurrentZoomLevel == "object" && this._map.getZoom() <= this.options.keepCurrentZoomLevel[1] && this._map.getZoom() >= this.options.keepCurrentZoomLevel[0];

if (this.options.keepCurrentZoomLevel)  would still be true in case of Array, so the if(..) still complicated. 


Or to have new `this.options.zoomOutsideRange` to store the range? 

      var zoomOutsideRange = typeof this.options.zoomOutsideRange == "object" && this._map.getZoom() <= this.options.zoomOutsideRange[1] && this._map.getZoom() >= this.options.zoomOutsideRange[0];

Then could just do... keepCurrentZoomLevel is back to just true/false

    if (this.options.keepCurrentZoomLevel || zoomOutsideRange )